src/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs

```
@@ -50,16 +50,43 @@ public TimeSpan PrivilegedProcessorTime
+                //
+                // ri_proc_start_abstime is absolute time which is the time since some reference point. the reference point
+                // usually is the system boot time but this is not necessary true in all versions of OSX. for example in Sierra 
+                // Mac OS 10.12 looks the reference point is not really the boot time. to be always accurate in our calculations
+                // we don’t assume the reference point to be the boot time and instead we calculate it by calling GetTimebaseInfo
+                // which is a wrapper for native mach_absolute_time(). that method returns the current time referenced to the needed start point
+                // then we can subtract the returned time from DateTime.UtcNow and we’ll get the exact reference point.
+                //
+                // The absolute time is measured by the bus cycle of the processor which is measured in nanoseconds multiplied 
+                // by some factor “numer / denom”. in most cases the factor is (1 / 1) but we have to get the factor and use it just 
+                // in case we run on a machine has different factor. to get the factor we call GetTimebaseInfo the wrapper for the 
+                // mach_timebase_info() which give us the factor. then we multiply the factor by the absolute time and the divide
+                // the result by 10^9 to convert it from nanoseconds to seconds.
+
                 EnsureState(State.HaveId);
+
+                uint numer, denom;
+                Interop.Sys.GetTimebaseInfo(out numer, out denom);
                 Interop.libproc.rusage_info_v3 info = Interop.libproc.proc_pid_rusage(_processId);
-                double seconds = info.ri_proc_start_abstime / (double)NanoSecondToSecondFactor;
+                double seconds;
+                long absoluteTime;
+
+                if (Interop.Sys.GetAbsoluteTime(out absoluteTime))
+                {
+                    // usually seconds will be negative
```

I'll change this to throw exception instead of falling back